lib/rules/brace-style.js

```diff
@@ -67,21 +67,31 @@ module.exports = function(context) {
                 previousToken = sourceCode.getTokenBefore(block);
                 curlyToken = sourceCode.getFirstToken(block);
                 curlyTokenEnd = sourceCode.getLastToken(block);
-                curlyTokensOnSameLine = curlyToken.loc.start.line === curlyTokenEnd.loc.start.line;
+                allOnSameLine = previousToken.loc.start.line === curlyToken.loc.start.line &&
```

Comment:
You should be able to remove the check in the middle and just compare previous to end.